### DISCUSSION TITLE
Trouble linking Jolt in Release mode

### DISCUSSION THREAD

**Discussion by Mahmoud1205** [jrouwe/JoltPhysics#1272] [Q&A] [answered]
I am getting multiple of these types of errors below..
`[build] lld-link: error: Jolt.dir\Release\CharacterVirtual.obj: is not a native COFF file. Recompile without /GL?`

Below are the relevant parts of my CMakeLists.txt file, it is not the whole file.
```cmake
set(CMAKE_C_COMPILER "clang")
set(CMAKE_CXX_COMPILER "clang++")

set(CMAKE_GENERATOR "Ninja" CACHE INTERNAL "" FORCE)

project(game)

set(BUILD_SHARED_LIBS OFF)
set(CMAKE_MSVC_RUNTIME_LIBRARY "MultiThreaded")

set(CMAKE_EXPORT_COMPILE_COMMANDS ON)

set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -O0 -std=c++20")
set(CMAKE_CXX_STANDARD 20)
set(CMAKE_CXX_STANDARD_REQUIRED ON)

# I renamed Jolt.lib to JoltRelease.lib
set(LIBS JoltRelease glfw3_mt)
target_link_libraries(${PROJECT_NAME} ${LIBS})
```
I suspected that this is because of LTO so I added `set(CMAKE_INTERPROCEDURAL_OPTIMIZATION ON)` to my CMakeLists but it is still giving the same error.

**Comment by jrouwe** [comment]
Can you try the `SET_INTERPROCEDURAL_OPTIMIZATION()` macro in your own CMakeLists.txt file instead?

Alternatively use the Jolt cmake option to turn LTO off:

```set(INTERPROCEDURAL_OPTIMIZATION OFF)```

**Reply by Mahmoud1205** [reply]
I am trying to build again after turning LTO off in Jolt but CMake is trying to use VS preview version which I don't have installed.
```
CMake Error at CMakeLists.txt:3 (project):
  Generator
    Visual Studio 17 2022
  could not find specified instance of Visual Studio:
    C:/Program Files/Microsoft Visual Studio/2022/Preview
-- Configuring incomplete, errors occurred!
```
The location of my CMake executable is `C:\Program Files\CMake\bin\cmake.exe`. It is not related to Visual Studio.
Sorry since this is not related to Jolt but I have not found any solutions for this online.

**Reply by jrouwe** [reply]
Sorry, I can't help you with that.

**Reply by Mahmoud1205** [reply]
I deleted the cache and it got fixed, but the errors are still there even after turning off LTO in Jolt.
I also tried to clean my project and rebuild it but it still shows these errors..

**Comment by jankrassnigg** [comment]
This can happen after a Visual Studio update.
1. Reboot Windows.
2. Clean the CMake cache, regenerate.
3. If you see these COFF errors, CLEAN the solution and then REBUILD.

**Reply by Mahmoud1205** [reply]
I tried this and it still didn't work.

**Comment by Mahmoud1205** [comment, marked as answer]
I moved from CMake to Visual Studio solutions and the error is no longer there.
